### DISCUSSION TITLE
Rootless docker

### DISCUSSION THREAD

**Discussion by guidot** [Budibase/budibase#6582] [Help]
I'm setting up my first docker install on a debian 11 server using the docker repos.
I configured docker to run rootless according to the [docs](https://docs.docker.com/engine/security/rootless/).

The socket is configured per environment variable to `DOCKER_HOST=unix:///run/user/998/docker.sock`.

`budibase-docker-watchtower-service` complains:
`level=error msg="Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?"`
It is running, I checked.

So I looked into `docker-compose.yaml` and changed the watchtower-service to
```
volumes:
      - /run/user/998/docker.sock:/run/user/998/docker.sock
```
But the error remains the same.
Is the socket hardcoded somewhere? Or do I need to set it in some other place?

Thanks for any help!

**Comment by daviddpd** [comment]
Not exactly sure on Debian 11, I'm using EL 7/8 (Rocky 8.6 right now, and really prefer FreeBSD ... and hate docker .... )  ... 

But this is typically meaning that docker isn't running.  I'm running as root (as I'm just evaluating - not the production setup )  - in EL8 with systemd - what usually fixes it :  

```
systemctl enable docker.socket
systemctl enable docker.service
systemctl restart docker.socket
systemctl restart docker.service
```

For running rootless ... (as non root user ) - I'd say you have to start docker.service and docker.socket as that user.  The best practices for this - I do not know.

**Comment by guidot** [comment]
Docker _is_ running and budibase is working normally.
The error message is coming from the watchtower, which AFAIK only takes care of updates.

**Comment by guidot** [comment]
I created an issue here  #6698
